sklearn/feature_selection/_rfe.py

```diff
@@ -185,8 +191,10 @@ def _fit(self, X, y, step_score=None):
         n_features = X.shape[1]
         if self.n_features_to_select is None:
             n_features_to_select = n_features // 2
+        elif 0.0 < self.n_features_to_select < 1.0:
```

Comment:
I think we should explicitly check for type and include `1.0`

```suggestion
        elif (isinstance(self.n_features_to_select, numbers.Real) and 
              0.0 < self.n_features_to_select <= 1.0):
```

The reason is to make sure that `1.0` means "select everything" and does not mean "select one feature". We should also test for this case. (Although I agree setting `1.0` is a little strange in this context.)